GoogleForCreators/web-stories-wp
# Supporting larger query for fetching Taxonomy Terms

## Context

As part of #11623, the endpoint for retrieving data about the Story would only return a maximum of ten taxonomy terms. This caused issues when one would add more than ten taxonomies within the editor. 

This is primarily a PHP change, would look for feedback from @swissspidy and @spacedmonkey as this is my first time touching PHP in 7+ years 😆  



## Summary



Below is a screenshot of this feature working as intended with the fix:
<img width="403" alt="Screen Shot 2022-06-06 at 4 31 32 PM" src="https://user-images.githubusercontent.com/7110244/172244623-c0f67a3f-f534-45aa-bc33-e382668f10dd.png">


## Relevant Technical Choices

This increases the `per_page` from the default of 10 to 100 in the Stories_Base_Controller file. The default was ten, and if one has more than ten terms, the UI of the story editor does not reflect all of the taxonomies.



## To-do


None.

## User-facing changes



This will improve Taxonomies and allow for more than ten selected tags or categories to be viewed in the Editor without losing assigned tags over that number.

## Testing Instructions






This PR can be tested by following these steps:

1. Create a new story
2. Assign it in any space (All Stories, or Editor) more than ten tags, categories, or custom taxonomies
3. Publish and/or update your changes
4. Go back to All Stories, you should see the tags, and categories reflect the real number
5. Go back to the Story Editor for this story, in Document > Taxonomies, you should see all categories and taxonomies supported (up to 100).

You can also follow the guidance left on #11623 to replicate (just make sure to use more than ten tags or categories to replicate the problem).

## Reviews

### Does this PR have a security-related impact?


No.

### Does this PR change what data or activity we track or use?


No.

### Does this PR have a legal-related impact?


No.

## Checklist




---



Fixes #11623
